src/city/health.c

```diff
@@ -400,22 +400,20 @@ int city_health_get_global_sickness_level(void)
         return SICKNESS_LEVEL_LOW;
     }
 
-    int global_rating = building_sickness_level / building_number;
     int global_sickness_level = SICKNESS_LEVEL_LOW;
+    int current_sickness = (calc_adjust_with_percentage(global_sickness, 80) + calc_adjust_with_percentage(max_sickness, 20)) / 2;
```

Comment:
You got the variable names wrong.
I mean, technically it was me, but you need to get them right. 

Also,  keep the `int global_rating` line and delete the `int global_sickness_level` instead.

```suggestion
    int current_sickness = (calc_adjust_with_percentage(global_rating, 80) + calc_adjust_with_percentage(max_sickness_level, 20)) / 2;
```

Reply:
OK, fixing it now, I'm so sorry. :(

Reply:
Well... it doesn't work.

One building is plagued, and the advisor says that "Health is almost perfect in the city" (which is true somewhat), and "Disease is a rare occurrence in the city". :sweat_smile:

Reply:
Instead of 80/20, try 50/50 then.

Reply:
> Instead of 80/20, try 50/50 then.

It does not work too. The lines stay the same (Health almost perfect / disease rare occurrence) even when a building is plagued.

__________________________

I think we should discuss this, with this particular case below:

We have two choices here:

**- The Advisor lines change based on the highest sickness value in the city.**

**Pros**: The player is informed as soon as sickness gets high somewhere in the city. No discrepancies between the health advisor rating and the sickness report, if at least one building have a high sickness or is plagued.

**Cons**: Only one sickness value defines what the Advisor says (the city health can be perfect, but is rated average or bad because of only one building having sickness >60, or >90, or =100).

**- The Advisor lines change based average sickness in the city, somewhat weighted with highest sickness value in the city .**

**Pros**: Health advisor is more "in line" with the global health in the city, even if only a few building has a high sickness level.

**Cons**: There's a discrepancy between the Health advisor rating and the sickness report (City Health is perfect, but "_many diseases are spreading in some areas_", or a "_building is plagued_" messages).

______________________________________
I think there's an alternate solution: rather than forcing Advisor lines, why not simply rewording the sickness messages, to make clearer that some buildings gets high sickness because of a their "personal" low health rating.

For example:
- City health is perfect (so the Advisor says "City health is perfect", no forced lines), but one building have +60 sickness:
_{TR_ADVISOR_SICKNESS_LEVEL_MEDIUM, "Some infectious diseases have appeared in the city **where health is lacking**, but the situation is under control. A sufficiently high level of healthcare will mitigate the impact of epidemics in the city."},_

- City health is perfect, but at least one building with +90 sickness:
{TR_ADVISOR_SICKNESS_LEVEL_HIGH, "Infectious diseases spread in **unhealthy** areas of the city. If the situation is not remedied soon, some buildings may be quarantined. Greater access to healthcare would effectively prevent epidemics from breaking out."},

- City health is perfect, but at least one building is plagued:
 { TR_ADVISOR_SICKNESS_LEVEL_PLAGUE, "The plague has come to **unhealthier** areas of the city! Many residents fall ill and buildings have been quarantined or burned to prevent the further spread of disease. What few doctors and surgeons are available have been requisitioned to decontaminate the most afflicted places." },
 _________________________________
We can get rid of the forced lines, and if a lot of buildings have a high sickness, it means the global health rating is bad anyway, so the Advisor lines will changed by themselves.